backend/lucene/src/main/java/org/hibernate/search/backend/lucene/types/codec/impl/LuceneFieldCodec.java

```diff
@@ -39,8 +40,8 @@
 	 * @param absoluteFieldPath The path of the field whose value is assumed to be stored.
 	 * @param collector A collector of absolute field paths to be extracted from the document.
 	 */
-	default void contributeStoredFields(String absoluteFieldPath, Consumer<String> collector) {
-		collector.accept( absoluteFieldPath );
+	default void contributeStoredFields(String absoluteFieldPath, Set<String> nestedDocumentPaths, BiConsumer<String, Set<String>> collector) {
```

Comment:
I'm not sure why you pass a `nestedDocumentPaths` here and expected the callback to pass it back to the collector.
Wouldn't it be simpler to just not mention the `nestedDocumentPaths` here, and expect the called to add them to the field visitor builder through a dedicated, separate method (e.g. `addNestedDocumentPaths`)?

Reply:
You're definitely right. Done!